src/oidc_child/oidc_child.c

```diff
@@ -0,0 +1,1396 @@
+
+    if (get_access_token) {
+        DEBUG(SSSDBG_TRACE_ALL, "access_token: [%s].\n",
+                                dc_ctx->access_token_str);
+        DEBUG(SSSDBG_TRACE_ALL, "id_token: [%s].\n", dc_ctx->id_token_str);
+
+        if (dc_ctx->jwks_uri != NULL) {
+            ret = verify_token(main_ctx, dc_ctx);
+            if (ret != EOK) {
+                DEBUG(SSSDBG_OP_FAILURE, "Failed to verify tokens.\n");
+                goto fail;
+            }
+        }
+
+        ret = get_userinfo(dc_ctx);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to get userinfo.\n");
+            goto fail;
+        }
+
+        dc_ctx->userinfo = json_loads(dc_ctx->http_data, 0, &json_error);
+        if (dc_ctx->userinfo == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "Failed to parse userinfo data on line [%d]: [%s].\n",
+                  json_error.line, json_error.text);
+            goto fail;
+        }
+
+        if (dc_ctx->access_token_payload != NULL) {
+            DEBUG(SSSDBG_TRACE_ALL, "access_token payload: [%s].\n",
+                  json_dumps(dc_ctx->access_token_payload, 0));
+
+            DEBUG(SSSDBG_CONF_SETTINGS, "User Principal: [%s].\n", json_string_value(json_object_get(dc_ctx->access_token_payload, "upn")));
+            DEBUG(SSSDBG_CONF_SETTINGS, "User oid: [%s].\n", json_string_value(json_object_get(dc_ctx->access_token_payload, "oid")));
+            DEBUG(SSSDBG_CONF_SETTINGS, "User sub: [%s].\n", json_string_value(json_object_get(dc_ctx->access_token_payload, "sub")));
+        }
+
+        if (dc_ctx->id_token_payload != NULL) {
+            DEBUG(SSSDBG_TRACE_ALL, "id_token payload: [%s].\n",
+                  json_dumps(dc_ctx->id_token_payload, 0));
+
+            DEBUG(SSSDBG_CONF_SETTINGS, "User Principal: [%s].\n", json_string_value(json_object_get(dc_ctx->id_token_payload, "upn")));
+            DEBUG(SSSDBG_CONF_SETTINGS, "User oid: [%s].\n", json_string_value(json_object_get(dc_ctx->id_token_payload, "oid")));
+            DEBUG(SSSDBG_CONF_SETTINGS, "User sub: [%s].\n", json_string_value(json_object_get(dc_ctx->id_token_payload, "sub")));
+        }
+
+        DEBUG(SSSDBG_CONF_SETTINGS, "userinfo: [%s].\n",
+                                    json_dumps(dc_ctx->userinfo, 0));
+
+        user_identifier = get_user_identifier(dc_ctx, dc_ctx->userinfo,
+                                              user_identifier_attr);
+        if (user_identifier == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to get user identifier.\n");
+            goto fail;
+        }
+
+        DEBUG(SSSDBG_CONF_SETTINGS, "User identifier: [%s].\n",
+                                    user_identifier);
+
+        fprintf(stdout,"%s", user_identifier);
+        fflush(stdout);
+    }
+
+    DEBUG(SSSDBG_IMPORTANT_INFO, "oidc_child finished successful!\n");
+    close(STDOUT_FILENO);
+    talloc_free(main_ctx);
+    curl_global_cleanup();
+    return EXIT_SUCCESS;
+
+fail:
```

Comment:
There is lots of duplication here. Using `done` and `ret` would be better.

Reply:
fixed